### PR TITLE
Exposes part.executedAt and piece.type in REST data

### PR DESCRIPTION
**Pull request by KvelaGorrrrnio**
The PR has the following changes:
- Event interfaces and enums have a more consistent format.
- Introduces TypedEvent with type and timestamp.
- PartDto.executedAt is added.
- PieceDto.type is added.
- Part.playedDuration is added (also on PartDto).
- Rundown.activate now resets the rundown.

### CODE REVIEW

File: src/model/entities/test/part.spec.ts [hunk 1]
```diff
@@ -18,6 +18,76 @@ describe('Part', () => {
...
+    afterEach(() => jest.useRealTimers())
+
+    it('sets part to be on air', () => {
+      const testee: Part = new Part({} as PartInterface)
+
+      testee.putOnAir()
+
+      const result: boolean = testee.isOnAir()
+      expect(result).toBe(true)
+    })
```
Comment (LindvedKrvang) [inline src/model/entities/test/part.spec.ts:31]: I think this is one of the few times more than one assertions would be nice.
A simple `expect(testee.isOnAir()).toBe(false)` before we call the implementation just to verify that the Part wasn't initialized to be onAir, which would make it possible for `putOnAir` to not put the Part on air without this test noticing.

File: src/model/enums/rundown-event-type.ts [hunk 2]
```diff
@@ -1,10 +1,10 @@
 export enum RundownEventType {
-  ACTIVATE = 'ACTIVATE',
-  DEACTIVATE = 'DEACTIVATE',
+  ACTIVATED = 'ACTIVATED',
+  DEACTIVATED = 'DEACTIVATE',
+  DELETED = 'DELETED',
   RESET = 'RESET',
-  TAKE = 'TAKE',
+  TAKEN = 'TAKE',
```
Comment (LindvedKrvang) [inline src/model/enums/rundown-event-type.ts:6]: Please keep the name of the enum attribute consistent with the value of the enum attribute.
If you change one, then also change the other.

File: src/business-logic/services/rundown-event-builder-implementation.ts [hunk 3]
```diff
@@ -1,90 +1,96 @@
...
+      timestamp: Date.now(),
       rundownId: rundown.id,
       segmentId: rundown.getActiveSegment().id,
       partId: rundown.getActivePart().id,
       adLibPiece: {
         id: adLibPiece.id,
         name: adLibPiece.name,
-        start: adLibPiece.getExecutedAt(),
+        layer: '', // TODO: Properly fill out
+        start: Math.min(0, adLibPiece.getExecutedAt() - rundown.getActivePart().getExecutedAt()), // TODO: Properly fill out
```
[the remark this answers is not in the document]
Comment (KvelaGorrrrnio) [reply src/business-logic/services/rundown-event-builder-implementation.ts:83]: Correct. This was to have same interface as planned pieces.

File: src/model/entities/segment.ts [hunk 4]
```diff
@@ -52,6 +52,7 @@ export class Segment {
   }
 
   public setAsNext(): void {
+    this.reset()
```
[the remark this answers is not in the document]
Comment (KvelaGorrrrnio) [reply src/model/entities/segment.ts]: Thanks for the advice :) I have ditched it again :+1:

File: src/model/entities/test/entity-mock-factory.ts [hunk 5]
```diff
@@ -67,6 +67,7 @@ export class EntityMockFactory {
     return instance(mockedSegment)
   }
 
+  // TODO: Rename this to createSegmentMock (same for the other methods)
   public static createSegmentMockInstance(
```
[the remark this answers is not in the document]
Comment (KvelaGorrrrnio) [reply src/model/entities/test/entity-mock-factory.ts:71]: It was out of concern for you and the conflicts you might need to handle ;) But I can do it if you want me to :rocket:

File: src/model/enums/rundown-event-type.ts (hunk 2 above)
Comment (KvelaGorrrrnio) [reply src/model/enums/rundown-event-type.ts]: Is changed.

File: src/model/entities/test/entity-mock-factory.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (RasmusAlbrektsen) [reply src/model/entities/test/entity-mock-factory.ts:71]: Also it's Shift + F6 if you use Mac (the superior thing you can develop on).

File: src/business-logic/services/rundown-timeline-service.ts [hunk 6]
```diff
@@ -98,7 +98,7 @@ export class RundownTimelineService implements RundownService {
     infinitePiecesAfter
       .filter((piece) => !infinitePiecesBefore.includes(piece))
       .forEach((piece) => {
-        const infinitePieceAddedEvent: InfiniteRundownPieceAddedEvent =
+        const infinitePieceAddedEvent: RundownInfinitePieceAddedEvent =
 					this.rundownEventBuilder.buildInfiniteRundownPieceAddedEvent(rundown, piece)
```
Comment (andr9528) [inline src/business-logic/services/rundown-timeline-service.ts:102]: Am i right in saying that some linting is missing the line being commented on? 
That line looks to be indented way to much. Might only be looking weird on GitHub.

File: src/model/entities/test/entity-mock-factory.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (andr9528) [reply src/model/entities/test/entity-mock-factory.ts:71]: For me it is (Ctrl + R) + (Ctrl + R), as that is what i have added, and is use to from Visual Studio :)

File: src/model/entities/test/rundown.spec.ts [hunk 7]
```diff
@@ -2029,4 +2029,30 @@ describe('Rundown', () => {
       })
     })
   })
+
+  describe('activate', () => {
+
+    it('resets all segments', () => {
+      const mockedSegment1: Segment = EntityMockFactory.createSegmentMockInstance()
+      const mockedSegment2: Segment = EntityMockFactory.createSegmentMockInstance()
+      const mockedSegment3: Segment = EntityMockFactory.createSegmentMockInstance()
```
[the remark this answers is not in the document]
Comment (KvelaGorrrrnio) [reply src/model/entities/test/rundown.spec.ts:2038]: Primarily for readability. By doing it with arrays, there will be a lot of boilerplate code that makes it harder to see what is happening. The test would look like the following.
```typescript
    it('resets all segments', () => {
      const mockedSegments: Segment[] = [...Array(3)].map(() => EntityMockFactory.createSegmentMockInstance())
      const segments: Segment[] = mockedSegments.map(mockedSegment => instance(mockedSegment))
      const testee: Rundown = new Rundown({
        segments,
        isRundownActive: false,
      } as RundownInterface)

      testee.activate()

      mockedSegments.forEach(mockedSegment => verify(mockedSegment.reset()).once())
    })
```

File: src/model/entities/test/entity-mock-factory.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (KvelaGorrrrnio) [reply src/model/entities/test/entity-mock-factory.ts]: Shift + Fn + F6 if you have disabled the function keys 😛
And I have renamed them.

File: src/business-logic/services/rundown-timeline-service.ts (hunk 6 above)
Comment (andr9528) [reply src/business-logic/services/rundown-timeline-service.ts:102]: I'd also put it on the same line, unless it gets past the max line width, which i think this one might. 
If too long, i'd line break similar to this, but not have the second piece indentet as much as it is here.

File: src/model/entities/test/rundown.spec.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (andr9528) [reply src/model/entities/test/rundown.spec.ts]: Might just be me, but apart from the first line, it looks way easier to read what this test does.

**Review by LindvedKrvang — APPROVED** [review]

**Review by RasmusAlbrektsen — APPROVED** [review]

**Pull request merged**